server/sonar-server/src/test/java/org/sonar/server/component/es/ProjectMeasuresIndexTest.java

```diff
@@ -118,9 +130,35 @@ public void filter_on_several_metrics() {
     assertThat(result).containsExactly("P2");
   }
 
+  @Test
+  public void return_only_authorized_projects() throws Exception {
```

Comment:
this test checks different use-cases at the same time. It should be split in different tests.